packages/belos/src/BelosGmresPolySolMgr.hpp

```diff
@@ -51,20 +51,8 @@
 
 #include "BelosLinearProblem.hpp"
 #include "BelosSolverManager.hpp"
-
 #include "BelosGmresPolyOp.hpp"
-#include "BelosGmresIteration.hpp"
-#include "BelosBlockGmresIter.hpp"
-#include "BelosDGKSOrthoManager.hpp"
-#include "BelosICGSOrthoManager.hpp"
-#include "BelosIMGSOrthoManager.hpp"
-#include "BelosStatusTestMaxIters.hpp"
-#include "BelosStatusTestGenResNorm.hpp"
-#include "BelosStatusTestImpResNorm.hpp"
-#include "BelosStatusTestCombo.hpp"
-#include "BelosStatusTestOutputFactory.hpp"
-#include "BelosOutputManager.hpp"
-#include "Teuchos_BLAS.hpp"
+#include "BelosSolverFactory_Generic.hpp"
```

Comment:
It's not a good idea to include solver factory stuff in a specific solver's header file.  That could cause circular references.  For examples of how I did this with Tpetra, see `belos/tpetra/src/solvers`.  I can explain in more detail over the phone if you like.